espressif/efuse/emulate_efuse_controller_base.py

```diff
@@ -28,15 +28,18 @@ def __init__(self, efuse_file=None, debug=False):
         self.efuse_file = efuse_file
         if self.efuse_file:
             try:
-                self.mem = BitString(open(self.efuse_file, 'a+b'), length=self.REGS.EFUSE_MEM_SIZE * 8)
-            except ValueError:
+                self.mem = BitStream(
+                    bytes = open(self.efuse_file, "rb").read(),
+                    length=self.REGS.EFUSE_MEM_SIZE * 8,
+                )
+            except (ValueError, FileNotFoundError):
```

Comment:
For example, `FileNotFoundError` is not known on Python 2.7. I expect the PR check will fail on this line.

Reply:
Thank you for bringing this up. I realize I need to do more thinking about this, so I will and come back with some ideas.

Reply:
Probably something similar could help:
```python
try:
    FileNotFoundError
except NameError:
    FileNotFoundError = IOError
```

Reply:
This is a great solution. I'm going to pull it in and see what CI says.